consensus/ibft/src/main/java/tech/pegasys/pantheon/consensus/ibft/statemachine/IbftBlockHeightManager.java

```diff
@@ -147,36 +151,31 @@ public void roundExpired(final RoundExpiry expire) {
   public void handleProposalMessage(final SignedData<ProposalPayload> msg) {
     LOG.info("Received a Proposal message.");
     actionOrBufferMessage(
-        msg,
-        () -> currentRound.handleProposalMessage(msg),
-        (roundstate) -> roundstate.setProposedBlock(msg));
+        msg, currentRound::handleProposalMessage, (roundstate) -> roundstate.setProposedBlock(msg));
   }
 
   public void handlePrepareMessage(final SignedData<PreparePayload> msg) {
     LOG.info("Received a prepare message.");
     actionOrBufferMessage(
-        msg,
-        () -> currentRound.handlePrepareMessage(msg),
-        (roundstate) -> roundstate.addPrepareMessage(msg));
+        msg, currentRound::handlePrepareMessage, (roundstate) -> roundstate.addPrepareMessage(msg));
   }
 
   public void handleCommitMessage(final SignedData<CommitPayload> msg) {
     LOG.info("Received a commit message.");
     actionOrBufferMessage(
-        msg,
-        () -> currentRound.handleCommitMessage(msg),
-        (roundstate) -> roundstate.addCommitMessage(msg));
+        msg, currentRound::handleCommitMessage, (roundstate) -> roundstate.addCommitMessage(msg));
   }
 
-  private void actionOrBufferMessage(
-      final SignedData<? extends Payload> msg,
-      final Runnable inRoundHandler,
+  private <T extends Payload> void actionOrBufferMessage(
+      final SignedData<T> msg,
+      final Consumer<SignedData<T>> inRoundHandler,
       final Consumer<RoundState> buffer) {
```

Comment:
with inRoundHandler accepting the message but the buffer lambda having the msg already inside it, feels mismatched. Need to make them both take msg or neither. I'm leaning towards both and making buffer be BiConsumer<RoundState, SignedData<T>>

Reply:
Done.